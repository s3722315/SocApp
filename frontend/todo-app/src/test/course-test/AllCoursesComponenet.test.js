import React from 'react'
import AllCourseComponent from '../../components/course-list/AllCoursesComponent.jsx'
import AllCourseComponentMock from './AllCourseComponentMock.jsx'
import { mount, shallow, render } from 'enzyme'


describe('AllCourseComponent', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<AllCourseComponent debug />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('AllCourseComponentMock', () => {
  it('enrolling from all courses list', () => {
    const component = mount(<AllCourseComponentMock debug />);
    
    component
    .find('#all-enroll-1')
    .simulate('click');

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('unenrolling from all courses list', () => {
    const component = mount(<AllCourseComponentMock debug />);
    
    component
    .find('#all-unenroll-4')
    .simulate('click');

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('unenrolling from my courses list', () => {
    const component = mount(<AllCourseComponentMock debug />);
    
    component
    .find('#my-unenroll-4')
    .simulate('click');

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});