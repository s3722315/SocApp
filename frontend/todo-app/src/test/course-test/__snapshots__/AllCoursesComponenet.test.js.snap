// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllCourseComponent should render correctly in "debug" mode 1`] = `
<div>
  <div
    className="container"
  >
    <h1>
      My Courses
    </h1>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Code
          </th>
          <th>
            Name
          </th>
          <th>
            Status
          </th>
          <th>
            Link
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <hr
      style={
        Object {
          "backgroundColor": "grey",
          "color": "grey",
          "height": 3,
        }
      }
    />
    <h1>
      All Courses
    </h1>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Code
          </th>
          <th>
            Name
          </th>
          <th>
            Status
          </th>
          <th>
            Link
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`AllCourseComponentMock enrolling from all courses list 1`] = `
<AllCoursesComponentMock
  debug={true}
>
  <div>
    <div
      className="container"
    >
      <h1>
        My Courses
      </h1>
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Code
            </th>
            <th>
              Name
            </th>
            <th>
              Status
            </th>
            <th>
              Link
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="4"
          >
            <td>
              bruh4
            </td>
            <td>
              bruh chad
            </td>
            <td>
              enrolled
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-warning"
                id="my-unenroll-4"
                onClick={[Function]}
              >
                Unenroll
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <hr
        style={
          Object {
            "backgroundColor": "grey",
            "color": "grey",
            "height": 3,
          }
        }
      />
      <h1>
        All Courses
      </h1>
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Code
            </th>
            <th>
              Name
            </th>
            <th>
              Status
            </th>
            <th>
              Link
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="1"
          >
            <td>
              bruh1
            </td>
            <td>
              bruh science
            </td>
            <td>
              available
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-success"
                id="all-enroll-1"
                onClick={[Function]}
              >
                Enroll
              </button>
            </td>
          </tr>
          <tr
            key="2"
          >
            <td>
              bruh2
            </td>
            <td>
              bruh physics
            </td>
            <td>
              unavailable
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              None
            </td>
          </tr>
          <tr
            key="3"
          >
            <td>
              bruh3
            </td>
            <td>
              bruh bro
            </td>
            <td>
              full
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              None
            </td>
          </tr>
          <tr
            key="4"
          >
            <td>
              bruh4
            </td>
            <td>
              bruh chad
            </td>
            <td>
              enrolled
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-warning"
                id="all-unenroll-4"
                onClick={[Function]}
              >
                Unenroll
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</AllCoursesComponentMock>
`;

exports[`AllCourseComponentMock unenrolling from all courses list 1`] = `
<AllCoursesComponentMock
  debug={true}
>
  <div>
    <div
      className="container"
    >
      <h1>
        My Courses
      </h1>
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Code
            </th>
            <th>
              Name
            </th>
            <th>
              Status
            </th>
            <th>
              Link
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="4"
          >
            <td>
              bruh4
            </td>
            <td>
              bruh chad
            </td>
            <td>
              enrolled
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-warning"
                id="my-unenroll-4"
                onClick={[Function]}
              >
                Unenroll
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <hr
        style={
          Object {
            "backgroundColor": "grey",
            "color": "grey",
            "height": 3,
          }
        }
      />
      <h1>
        All Courses
      </h1>
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Code
            </th>
            <th>
              Name
            </th>
            <th>
              Status
            </th>
            <th>
              Link
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="1"
          >
            <td>
              bruh1
            </td>
            <td>
              bruh science
            </td>
            <td>
              available
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-success"
                id="all-enroll-1"
                onClick={[Function]}
              >
                Enroll
              </button>
            </td>
          </tr>
          <tr
            key="2"
          >
            <td>
              bruh2
            </td>
            <td>
              bruh physics
            </td>
            <td>
              unavailable
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              None
            </td>
          </tr>
          <tr
            key="3"
          >
            <td>
              bruh3
            </td>
            <td>
              bruh bro
            </td>
            <td>
              full
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              None
            </td>
          </tr>
          <tr
            key="4"
          >
            <td>
              bruh4
            </td>
            <td>
              bruh chad
            </td>
            <td>
              enrolled
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-warning"
                id="all-unenroll-4"
                onClick={[Function]}
              >
                Unenroll
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</AllCoursesComponentMock>
`;

exports[`AllCourseComponentMock unenrolling from my courses list 1`] = `
<AllCoursesComponentMock
  debug={true}
>
  <div>
    <div
      className="container"
    >
      <h1>
        My Courses
      </h1>
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Code
            </th>
            <th>
              Name
            </th>
            <th>
              Status
            </th>
            <th>
              Link
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="4"
          >
            <td>
              bruh4
            </td>
            <td>
              bruh chad
            </td>
            <td>
              enrolled
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-warning"
                id="my-unenroll-4"
                onClick={[Function]}
              >
                Unenroll
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <hr
        style={
          Object {
            "backgroundColor": "grey",
            "color": "grey",
            "height": 3,
          }
        }
      />
      <h1>
        All Courses
      </h1>
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Code
            </th>
            <th>
              Name
            </th>
            <th>
              Status
            </th>
            <th>
              Link
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="1"
          >
            <td>
              bruh1
            </td>
            <td>
              bruh science
            </td>
            <td>
              available
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-success"
                id="all-enroll-1"
                onClick={[Function]}
              >
                Enroll
              </button>
            </td>
          </tr>
          <tr
            key="2"
          >
            <td>
              bruh2
            </td>
            <td>
              bruh physics
            </td>
            <td>
              unavailable
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              None
            </td>
          </tr>
          <tr
            key="3"
          >
            <td>
              bruh3
            </td>
            <td>
              bruh bro
            </td>
            <td>
              full
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              None
            </td>
          </tr>
          <tr
            key="4"
          >
            <td>
              bruh4
            </td>
            <td>
              bruh chad
            </td>
            <td>
              enrolled
            </td>
            <td>
              <button
                className="btn btn-success"
              >
                Go To
              </button>
            </td>
            <td>
              <button
                className="btn btn-warning"
                id="all-unenroll-4"
                onClick={[Function]}
              >
                Unenroll
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</AllCoursesComponentMock>
`;
